hack/pre-release/pre-release-common.sh

```
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+# Common functions and variables for pre-release scripts
+
+# Initialize script and project directories
+SCRIPT_DIR="$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)